compiler/rustc_middle/src/mir/interpret/pointer.rs

```diff
@@ -159,6 +159,14 @@ pub struct Pointer<Tag = AllocId> {
 
 static_assert_size!(Pointer, 16);
 
+// We want the `Debug` output to be readable as it is used by `derive(Debug)` for
+// all the Miri types.
+impl<Tag: Provenance> fmt::Display for Pointer<Tag> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        Provenance::fmt(self, f)
```

Comment:
Instead of duplicating the code, please make one of `Display` and `Debug` dispatch to the other (not sure which way makes more sense, but with the existing comment it probably makes most sense to have `Debug` call `Display`)

Reply:
Well, we won't need two different ones, so forwarding makes it easier for future readers to see that they are the same

Reply:
Ok. Now I have a question: do you to have instead for `Debug`:

```rust
write!(f, "{}", self)
```

?

Reply:
I would have assumed `fmt::Display::fmt(self, f)`

Tho now we have to decide whether this is actually what we wanted, as ui tests changed. The hex dumps were somewhat on purpose there.

Reply:
That works too!

Reply:
> It allows to have two different formatting depending on what you need like that.

The issue is that it is entirely unclear which style is used where -- both debug and display end up being user-visible in various situations, and when changing the code it becomes extremely hard to predict which one will change which output. IOW we can't actually meaningfully make them different, currently, since the consequences are so unclear. That is why I entirely *removed* many of the `Display` implementations -- then at least there is no ambiguity over which kind of formatting is used.

This PR adds back some of those instances which I removed; if that cannot be avoided I won't block it but then we can at least consistently use the pattern of forwarding one to the other.